### PR TITLE
🌈 adding marker props

### PR DESCRIPTION
**Pull request by jonasvdd**
Updated the description @jvdd 

TODO:
- [x] more tests (e.g. serialization)
- [x] update `basic_example` or create a new example @jvdd, as the basic example already is rather large.
   * side note: I would also put the *minimal code examples*, which are in a `<detail>` element in the `README` of the examples.
- [x] verify docs

### CODE REVIEW

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 1]
```diff
@@ -283,6 +289,7 @@ def _check_update_trace_data(
                 trace["y"] = [None]
                 trace["text"] = ""
                 trace["hovertext"] = ""
+                # TODO -> add marker kwargs here?
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:292]: Not really sure if this is necessary

Comment (jvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:292]: Nvm, giving it some more thought - quite sure that this is necessary

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 2]
```diff
@@ -318,6 +325,7 @@ def _check_update_trace_data(
                     name = name[: -len(self._suffix)]
             trace["name"] = name
 
+            # TODO -> extra logic needs to be intertwined here
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:328]: What do you mean @jonasvdd ?

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 3]
```diff
@@ -592,8 +621,11 @@ def _parse_get_trace_props(
         trace: BaseTraceType,
         hf_x: Iterable = None,
         hf_y: Iterable = None,
+        # TODO -> are these type hints correct
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:624]: Not really sure tbh..

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 4]
```diff
@@ -804,19 +858,13 @@ def _construct_hf_data_dict(
...
             "max_n_samples": max_n_samples,
             "default_n_samples": default_n_samples,
-            "x": dc.x,
-            "y": dc.y,
             "axis_type": axis_type,
             "downsampler": downsampler,
             "default_downsampler": default_downsampler,
-            "text": dc.text,
-            "hovertext": dc.hovertext,
+            **dc._asdict(),
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:867]: Clean!! :handshake:

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 5]
```diff
@@ -892,13 +942,7 @@ def add_trace(
...
             The original high frequency hovertext. If set, this has priority over the
             trace its ```hovertext`` argument.
-        check_nans: boolean, optional
-            If set to True, the trace's data will be checked for NaNs - which will be
-            removed. By default True.
-            As this is a costly operation, it is recommended to set this parameter to
-            False if you are sure that your data does not contain NaNs (or when the
-            downsampler can handle NaNs, e.g., EveryNthPoint). This should considerably
-            speed up the graph construction time.
+        TODO docs
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:945]: Why did you remove `check_nans` documentation?

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 6]
```diff
@@ -1319,7 +1377,14 @@ def construct_update_data(
...
 
         # 2. Create the additional trace data for the frond-end
-        relevant_keys = ["x", "y", "text", "hovertext", "name"]  # TODO - marker color
+        relevant_keys = [
+            "x",
+            "y",
+            "text",
+            "hovertext",
+            "name",
+            "marker",
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:1386]: TODO: test this

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 7]
```diff
@@ -337,6 +345,27 @@ def _check_update_trace_data(
...
+            else:
+                trace["marker"]["size"] = marker_size
+
+            marker_color = hf_trace_data.get("marker_color")
+            if isinstance(marker_color, (np.ndarray, pd.Series)):
+                trace["marker"]["color"] = self._to_hf_series(
+                    x=hf_trace_data["x"], y=marker_color
+                ).loc[s_res.index]
+            else:
+                trace["marker"]["color"] = marker_color
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:367]:
```py
            # Iterate over the high-frequency marker properties (stored in the 
            # _hf_data_container) and add resampled values to the trace (if necessary)
            hf_marker_props = ["size", "color"]
            for prop in hf_marker_props:
                marker_prop = hf_trace_data.get(f"marker_{prop}")
                if isinstance(marker_prop, (np.ndarray, pd.Series)):
                    trace["marker"][prop] = self._to_hf_series(
                        x=hf_trace_data["x"], y=marker_prop,
                    ).loc[s_res.index]
                else:
                    trace["marker"][prop] = marker_prop
```

Alternative implementation :arrow_up:
But tbh, I'm fine with keeping the original (more verbose) one

File: plotly_resampler/figure_resampler/figure_resampler_interface.py (hunk 5 above)
Comment (jonasvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:945]: Whoops, this was a small mistake, due to the merge conflict with `main` 🙈

**Comment by jvdd** [comment]
TODO: test serialization with `hf_marker_size` and `hf_marker_color`

**Comment by jonasvdd** [comment]
updated the description @jvdd

**Comment by jonasvdd** [comment]
test seem to work @jvdd, will perform a first review with updated code somewhere later today!

File: plotly_resampler/figure_resampler/figure_resampler_interface.py (hunk 3 above)
Comment (jonasvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:552]: Me neither, will just remove this comment; as this is a hidden method.

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 8]
```diff
@@ -970,12 +1026,12 @@ def add_trace(
...
                 self._print(f"[i] NOT resampling {trace['name']} - len={n_samples}")
-                for k in dc._fields:
-                    setattr(trace, k, getattr(dc, k))
+                trace._process_kwargs(**{k: getattr(dc, k) for k in dc._fields})
                 return super(AbstractFigureAggregator, self).add_traces(
                     [trace], **self._add_trace_to_add_traces_kwargs(trace_kwargs)
                 )
-        return super(AbstractFigureAggregator, self).add_traces(
+
+        return super(self._figure_class, self).add_traces(
```
Comment (jonasvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:1034]: Why do we use here`self.figure_class`?

Comment (jonasvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:1038]: Is this to reduce the double call of `add_traces`?

**Comment by jonasvdd** [comment]
Ready for your review @jvdd!

File: plotly_resampler/figure_resampler/figure_resampler_interface.py (hunk 8 above)
Comment (jvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:1038]: Could be! I believe you investigated this before..

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 9]
```diff
@@ -955,6 +1014,7 @@ def add_trace(
                 # We copy (by reference) all the non-data properties of the trace in
                 # the new trace.
                 trace = trace._props  # convert the trace into a dict
+                # TODO -> do not use this difference -> and use a list of keys
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:1017]: Create a separate issue for this?

**Review by jvdd — APPROVED** [review]
LGTM - still some work to do for extending the tests etc.

**Comment by jonasvdd** [comment]
@jvdd, ready for final review!
fyi, I looked at the serialization code; and it just serializes the `hf_data` object, with `hf_marker_size` and `hf_marker_color` being sub-properties of `hf_data` (highly similar to `hf_x` and `hf_y`). As such, I think that adding extra tests for marker property serializations are not necessary.

**Review by jvdd — APPROVED** [review]
LGTM - just have a second look at the only open comment above

Nice job :handshake:
